packages/xstate-react/src/useInterpret.ts

```diff
@@ -136,6 +136,10 @@ export function useInterpret<TMachine extends AnyStateMachine>(
 
   useEffect(() => {
     const rehydratedState = options.state;
+    // @ts-ignore
+    // This ensures that the interpreter recomputes the initial state
+    // TODO: figure out a better way to do this
+    delete service._initialState;
```

Comment:
@Andarist There's probably a better way to handle this for v4, wdyt?

Reply:
Actually, I'd say that this is an acceptable solution in v4. However, I wonder why the same thing here isn't enough.
https://github.com/statelyai/xstate/blob/40514d22ccaf7bd99318de1bdd9c931231fc8019/packages/core/src/interpreter.ts#L598

After all, we already call `.stop()` in the effect's cleanup.

Reply:
Because of this: https://github.com/statelyai/xstate/blob/40514d22ccaf7bd99318de1bdd9c931231fc8019/packages/core/src/interpreter.ts#L591-L594

That line never gets called if the interpreter was never even started, which is the case here. 🕵️